### PR TITLE
Issue #14814: refactor containsStaticField into iterative method

### PR DESCRIPTION
**Pull request by mohitsatr**
part of #14814 : containsStaticField in HiddenFieldCheck

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java [hunk 1]
```diff
@@ -670,13 +669,18 @@ public boolean containsInstanceField(String field) {
...
         public boolean containsStaticField(String field) {
-            return staticFields.contains(field)
-                    || parent != null
-                    && parent.containsStaticField(field);
+            FieldFrame currentParent = parent;
+            boolean parentContains = true;
+            while (currentParent != null) {
+                parentContains = parentContains && currentParent.staticFields.contains(field);
+                if (parentContains) {
+                    break;
```
Comment (mohitsatr) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java:679]: this condition can be moved into while loop, but doing that has 0.95 branch coverage ratio.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java:679]: We need to add tests to recover coverage.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java [hunk 2]
```diff
@@ -341,7 +341,6 @@ private void processVariable(DetailAST ast) {
             // local variable or parameter. Does it shadow a field?
             final DetailAST nameAST = ast.findFirstToken(TokenTypes.IDENT);
             final String name = nameAST.getText();
-
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java:344]: Please restore this empty line.
It is not related to target fix.

**Review by romani — CHANGES_REQUESTED** [review]
CI must be green.
Sorry that code base is over demanding

**Comment by romani** [comment]
Github, generate report for HiddenField/all-examples-in-one

**Comment by romani** [comment]
Please look at diff report for cases we have regression on, we should add them to our Inputs

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java (hunk 1 above)
Comment (mohitsatr) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java]: no need to worry about that now. simplied the logic

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java (hunk 2 above)
Comment (mohitsatr) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java]: done

**Comment by mohitsatr** [comment]
Github, generate report for HiddenField/all-examples-in-one

**Review by romani — APPROVED** [review]
Ok to merge.

Thanks a lot.

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**